### DISCUSSION TITLE
Text classification: Input Format of train/dev file

### DISCUSSION THREAD

**Discussion by hp52** [explosion/spaCy#7064] [Help: Coding & Implementations] [answered]
Hi,
I want to train a text classifier with the command

python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./dev.spacy

I don't find any clear information what format the train and dev file should have. I see there is a convert command that converts a file into the spacy format, but I don't see what input format it's converted from.

Could you indicate me what format I should use?

Thank you.

**Comment by svlandeg** [comment]
Hi!

If you created a config.cfg file, e.g. with [`init config`](https://spacy.io/api/cli#init-config), the default reader will be `Corpus` and the config will look something like this:
```

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 2000
gold_preproc = false
limit = 0
augmenter = null
```

[`Corpus`](https://spacy.io/api/corpus#config) takes binary [.spacy files](https://spacy.io/api/data-formats#binary-training) that you can create by storing the relevant annotations on `Doc` objects and serializing those with `DocBin`.

If you have datasets in another format, you can indeed use `convert` FIRST before running the training utility. https://spacy.io/api/cli#convert has more details, including the [formats](https://spacy.io/api/cli#converters) from which you can convert (`json`, `conll`, `ner`, `iob`). The output files will be binary .spacy files, that you can then use with the `Corpus` reader when training.

**Comment by adrianeboyd** [comment, marked as answer]
Hi, I think part of the confusion here is that none of the `spacy convert` converters other than `json` deal with `textcat` annotation, and just for `textcat` it doesn't make sense to go through the old `json` format. Since you just need to create `Doc` objects with the right annotation, it makes more sense to write a short custom conversion script for your data.

You can just create docs with the right cats by hand and then save them in `.spacy` format:

```python
import spacy
from spacy.tokens import DocBin

nlp = spacy.blank("en") # choose the right language/tokenizer here

docs = []
for text, cats in zip(corpus_texts, corpus_cats):
    doc = nlp.make_doc(text)
    doc.cats = cats
    docs.append(doc)

db = DocBin(docs=docs)
db.to_disk("train.spacy")
```

**Reply by hp52** [reply]
Thank you, this information was helpful.
